x/transfermiddleware/keeper/ibc.go

```diff
@@ -0,0 +1,112 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+	"github.com/notional-labs/banksy/v2/x/transfermiddleware/types"
+)
+
+func (keeper Keeper) hasParachainIBCTokenInfo(ctx sdk.Context, nativeDenom string) bool {
+	store := ctx.KVStore(keeper.storeKey)
+	return store.Has(types.GetKeyKeysParachainIBCTokenInfo(nativeDenom))
+}
+
+func (keeper Keeper) handleOverrideSendPacketTransferLogic(
+	ctx sdk.Context,
+	chanCap *capabilitytypes.Capability,
+	sourcePort string, sourceChannel string,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+	data []byte,
+) (sequence uint64, err error) {
+	var fungibleTokenPacketData transfertypes.FungibleTokenPacketData
+
+	err = keeper.cdc.Unmarshal(data, &fungibleTokenPacketData)
+	if err != nil {
+		return 0, err
+	}
+	sender, err := sdk.AccAddressFromBech32(fungibleTokenPacketData.Sender)
+	if err != nil {
+		return 0, err
+	}
+
+	// check if denom in fungibleTokenPacketData is native denom in parachain info and
+	parachainInfo := keeper.GetParachainIBCTokenInfo(ctx, fungibleTokenPacketData.Denom)
+	if parachainInfo.ChannelId != sourceChannel {
+		keeper.SendPacket(ctx, chanCap, sourcePort, sourceChannel, timeoutHeight, timeoutTimestamp, data)
```

Comment:
I think we should call the underlying keeper.ics4Wrapper.SendPacket instead of keeper.Sendpacket, and should return this func